### PR TITLE
Install config-visualizer-gui, move Python installation to basics

### PR DESCRIPTION
**Pull request by MakisH**
@fsimonis any comments/suggestions?

I have not tested this yet, I plan to test all updates together.

### CODE REVIEW

File: provisioning/install-config-visualizer.sh [hunk 1]
```diff
@@ -1,18 +1,13 @@
...
-    git clone --depth=1 --branch master https://github.com/precice/config-visualizer.git
-fi
-pip3 install --user -e config-visualizer
+# Install CLI dependencies
+sudo apt-get install -y graphviz
+# Install GUI depedencies
+sudo apt-get install -y libcairo2-dev libgirepository1.0-dev gcc libcairo2-dev pkg-config gir1.2-gtk-4.0
+
+# Get the config-visualizer from PIP
+pipx install precice-config-visualizer precice-config-visualizer-gui
```
Comment (fsimonis) [inline provisioning/install-config-visualizer.sh:10]: One package at a time.

```suggestion
pipx install precice-config-visualizer
pipx install precice-config-visualizer-gui
```

File: provisioning/install-config-visualizer.sh [hunk 2]
```diff
@@ -1,18 +1,13 @@
 #!/usr/bin/env bash
 set -ex
 
-# Get the config-visualizer from GitHub
-if [ ! -d "config-visualizer/" ]; then
-    git clone --depth=1 --branch master https://github.com/precice/config-visualizer.git
-fi
-pip3 install --user -e config-visualizer
+# Install CLI dependencies
+sudo apt-get install -y graphviz
```
Comment (fsimonis) [inline provisioning/install-config-visualizer.sh:5]: An alternative could be to install the package from the graphviz repo, or even add it to the VM to keep it reproducible.

**Comment by fsimonis** [comment]
We could also install the icon and the .desktop file for better integration.

https://raw.githubusercontent.com/precice/config-visualizer-gui/main/data/org.precice.config_visualizer.svg

https://gist.githubusercontent.com/fsimonis/a08c3771abf808b0534d658bcb563f90/raw/e091d78c24b04d0fc903b8de4909528628d22b7b/org.precice.configvisualizer.desktop

File: provisioning/install-config-visualizer.sh (hunk 2 above)
Comment (MakisH) [reply provisioning/install-config-visualizer.sh:5]: But it works as-is, right? I want to keep things simple here, while external repositories are sometimes not responding and cancelling the whole VM build (we already have this issue with the OpenFOAM PPA on SourceForge).

I see the reproducibility concern, but I think this reproducible images are not required in the VM.

**Comment by MakisH** [comment]
Added the Desktop file (thanks!), but not the picture. I am not sure where to add it, and the linked file does not refer to it.

See also https://github.com/precice/config-visualizer-gui/issues/3

File: .github/workflows/check-shell.yml [hunk 3]
```diff
@@ -10,3 +10,5 @@ jobs:
         uses: ludeeus/action-shellcheck@master
         with:
           ignore: Vagrantfile
+        env:
+          SHELLCHECK_OPTS: -e SC2016
```
Comment (MakisH) [inline .github/workflows/check-shell.yml:14]: @fsimonis I would prefer that we don't globally disable rules. Better add

```
# shellcheck disable=SC2016
```

above the respective line.

Comment (MakisH) [reply .github/workflows/check-shell.yml:14]: Or I just do that directly.

**Pull request merged**